docs/integration-tutorial.md

```diff
@@ -0,0 +1,212 @@
+
+This section is intended to be a first step when building a new project from scratch. If you would prefer to start from a preconfigured environment, feel free to skip ahead to the next section.
+
+```sh
+mkdir <project_title>
+cd project_title
+truffle init
+yarn add @truffle/contract
+yarn add @gnosis.pm/dex-contracts
+[optional] yarn add @gnosis.pm/util-contracts
+[optional] yarn add @openzeppelin/contracts@2.5.1
+```
+
+Make sure that you have a valid (truffle configuration)[https://www.trufflesuite.com/docs/truffle/reference/configuration] for the network you intend on interacting with.
+We will be using the common `truffle-config.js` found in most, if not all, gnosis smart contract projects.
+In order to use the common gnosis truffle configuration, you will need to install the optional package `@gnosis.pm/util-contracts` listed above.
+Furthermore, you will likely have to provide your own `INFURA_KEY`
+
+Note that, if you plan to be experimenting with a locally hosted development network, you will need to install additional "devDepencencies" to mirgrate the `BatchExchange` Smart Contracts. This will be covered in detail once we have successfully confirmed our ability to interact with the existing mainnet smart contracts.
+
+## Interaction with Batch Exchange
+
+Assuming you were successful with optionally configuring your own project from the section above, we will continue here from a pre-configured environment obtained and installed as follows:
+
+```sh
+git clone git@github.com:bh2smith/dex-integration-tutorial.git
+cd dex-integration-tutorial
+yarn
+```
+
+This should put us in the same place as has having completed the project initialization steps independantly.
+We are now prepared to start scripting interactions with the Gnosis Protocol.
+To test this run the `exchange_interaction` script via
+
+Note that before attempting to execute any of these truffle scripts, you will have to export your own `INFURA_KEY`!
+
+```sh
+export INFURA_KEY=<your infura key>
+```
+
+For less repitiion, this can also be included directly in the truffle configuration file at [line 17](https://github.com/bh2smith/dex-integration-tutorial/blob/44320ed3345ddb72ea5f0884460e8e1789ad48fe/truffle-config.js#L17). Just replace the existing one after the `||`. Be carefull not to commit your infura key!
+
+```sh
+npx truffle exec scripts/exchange_interaction.js --network rinkeby
+```
+
+and observe the following logs:
+
+```
+Using network 'rinkeby'.
+
+Aquired Batch Exchange 0xC576eA7bd102F7E476368a5E98FA455d1Ea34dE2
+Current Batch 5308007
+```
+
+This script simply aquires the BatchExchange contract deployed at the appropriate network and prints the current batch index.
+
+A few important lines used throughout such integration are are the following import statements used for acquiring the Batch Exchange contract artifacts according to the correct network.
+
+```js
+const Contract = require("@truffle/contract")
+const BatchExchange = Contract(
```

Comment:
We changed how the migration works. Sorry, this is outdated. Truffle contracts is no longer required and you can simply do const BatchExchange = artifacts.require after running yarn build. Check out the dex-liquidity-provision repo as an example.